src/algebra/group/conj.lean

```diff
@@ -45,8 +70,140 @@ calc is_conj a 1 ↔ is_conj 1 a : ⟨is_conj_symm, is_conj_symm⟩
+    exact hc
+  end⟩⟩
+
+namespace is_conj
+/- This small quotient API is largely copied from the API of `associates`;
+where possible, try to keep them in sync -/
+
+/-- The setoid of the relation `is_conj` iff there is a unit `u` such that `u * x = y * u` -/
+protected def setoid (α : Type*) [monoid α] : setoid α :=
+{ r := is_conj, iseqv := ⟨is_conj_refl, λa b, is_conj_symm, λa b c, is_conj_trans⟩ }
+
+end is_conj
+
+local attribute [instance, priority 100] is_conj.setoid
+
+/-- The quotient type of conjugacy classes of a group. -/
+def conj_classes (α : Type*) [monoid α] : Type* :=
+quotient (is_conj.setoid α)
+
+namespace conj_classes
+
+section monoid
+variables [monoid α] [monoid β]
+
+/-- The canonical quotient map from a monoid `α` into the `conj_classes` of `α` -/
+protected def mk {α : Type*} [monoid α] (a : α) : conj_classes α :=
+⟦ a ⟧
+
+instance : inhabited (conj_classes α) := ⟨⟦1⟧⟩
+
+theorem mk_eq_mk_iff_is_conj {a b : α} :
+  conj_classes.mk a = conj_classes.mk b ↔ is_conj a b :=
+iff.intro quotient.exact quot.sound
+
+theorem quotient_mk_eq_mk (a : α) : ⟦ a ⟧ = conj_classes.mk a := rfl
+
+theorem quot_mk_eq_mk (a : α) : quot.mk setoid.r a = conj_classes.mk a := rfl
+
+theorem forall_is_conj {p : conj_classes α → Prop} :
+  (∀a, p a) ↔ (∀a, p (conj_classes.mk a)) :=
+iff.intro
+  (assume h a, h _)
+  (assume h a, quotient.induction_on a h)
+
+theorem mk_surjective : function.surjective (@conj_classes.mk α _) :=
+forall_is_conj.2 (λ a, ⟨a, rfl⟩)
+
+instance : has_one (conj_classes α) := ⟨⟦ 1 ⟧⟩
+
+theorem one_eq_mk_one : (1 : conj_classes α) = conj_classes.mk 1 := rfl
+
+lemma exists_rep (a : conj_classes α) : ∃ a0 : α, conj_classes.mk a0 = a :=
+quot.exists_rep a
+
+/-- A `monoid_hom` maps conjugacy classes of one group to conjugacy classes of another. -/
+def map (f : α →* β) : conj_classes α → conj_classes β :=
+quotient.lift (conj_classes.mk ∘ f) (λ a b ab, mk_eq_mk_iff_is_conj.2 (f.map_is_conj ab))
+
+instance [fintype α] [decidable_rel (is_conj : α → α → Prop)] :
+  fintype (conj_classes α) :=
+quotient.fintype (is_conj.setoid α)
+
+end monoid
+
+section comm_monoid
+variable [comm_monoid α]
+
+lemma mk_injective : function.injective (@conj_classes.mk α _) :=
+λ _ _, (mk_eq_mk_iff_is_conj.trans is_conj_iff_eq).1
+
+lemma mk_bijective : function.bijective (@conj_classes.mk α _) :=
+⟨mk_injective, mk_surjective⟩
+
+/-- The bijection between a `comm_group` and its `conj_classes`. -/
+def mk_equiv : α ≃ conj_classes α :=
+⟨conj_classes.mk, quotient.lift id (λ (a : α) b, is_conj_iff_eq.1), quotient.lift_mk _ _,
+  begin
+    rw [function.right_inverse, function.left_inverse, forall_is_conj],
+    intro x,
+    rw [← quotient_mk_eq_mk, ← quotient_mk_eq_mk, quotient.lift_mk, id.def],
+  end⟩
+
+end comm_monoid
+end conj_classes
+
+section monoid
+
+variables [monoid α]
+
+/-- Given an element `a`, `conjugates a` is the set of conjugates. -/
+def conjugates (a : α) : set α := {b | is_conj a b}
```

Comment:
Should this be called `conjugates_of a`? I find myself reading `conjugates a` as though `conjugates` were a verb. Maybe see if anyone on zulip has an opinion.